### PR TITLE
Add contained_fastlane? to Helper.rb

### PR DESCRIPTION
**Pull request by KrauseFx**
This is for https://github.com/fastlane/fastlane/pull/6858

### CODE REVIEW

File: fastlane_core/lib/fastlane_core/helper.rb [hunk 1]
```diff
@@ -39,6 +39,12 @@ def self.bundler?
       return false
     end
 
+    # Do we run from a bundled fastlane, which contains Ruby and OpenSSL?
+    # Usually this means the fastlane directory is ~/.fastlane/bin/
+    def self.contained_fastlane?
+      ENV["FASTLANE_IS_BUNDLED"].to_s.length > 0
```
Comment (milch) [inline fastlane_core/lib/fastlane_core/helper.rb]: See [this comment](https://github.com/fastlane/fastlane/pull/6858/files#r86898616)

**Review by milch — CHANGES_REQUESTED** [review]

File: fastlane_core/lib/fastlane_core/helper.rb [hunk 2]
```diff
@@ -39,6 +39,13 @@ def self.bundler?
       return false
     end
 
+    # Do we run from a bundled fastlane, which contains Ruby and OpenSSL?
+    # Usually this means the fastlane directory is ~/.fastlane/bin/
+    # We set this value via the environment variable `SELF_CONTAINED`
+    def self.contained_fastlane?
+      ENV.key?("SELF_CONTAINED")
```
Comment (milch) [inline fastlane_core/lib/fastlane_core/helper.rb]: I'd rather we explicitly check `ENV["SELF_CONTAINED"].to_s == "true"` here

**Review by milch — CHANGES_REQUESTED** [review]

File: fastlane_core/lib/fastlane_core/helper.rb (hunk 2 above)
Comment (KrauseFx) [inline fastlane_core/lib/fastlane_core/helper.rb]: In general
```
ENV["SELF_CONTAINED"].to_s == "true
```
doesn't work when the user uses
```
export SELF_CONTAINED="1"
```
however for this case it's good enough

**Review by milch — APPROVED** [review]

**Comment by milch** [comment]
🔥

**Pull request merged**